src/analysis/retail/warrior/fury/modules/spells/SlayerExecute.tsx

```diff
@@ -0,0 +1,193 @@
+      isGreaterThan: {
+        minor: 0,
+        average: 5,
+        major: 12,
+      },
+      style: ThresholdStyle.NUMBER,
+    };
+  }
+
+  onMarkedForExecutionStackChange(
+    event: ApplyDebuffEvent | ApplyDebuffStackEvent | RemoveDebuffStackEvent | RemoveDebuffEvent,
+  ) {
+    this.markedForExecutionStacks = currentStacks(event);
+  }
+
+  updateRemoveBuffStack(event: RemoveBuffStackEvent) {
+    this.lastSuddenDeathRemoveBuffStackTimestamp = event.timestamp;
+  }
+
+  onSuddenDeathStackChange(event: ApplyBuffEvent | RefreshBuffEvent) {
+    // 2->1 stacks triggers refresh buff event, but doesn't extend duration
+    // so ignore those
+    // but we still want to listen to refreshbuff because
+    // 2->2 stacks triggers a refresh event and extends duration
+    if (
+      event.type === EventType.RefreshBuff &&
+      event.timestamp - this.lastSuddenDeathRemoveBuffStackTimestamp < 100
+    ) {
+      return;
+    }
+    this.suddenDeathExpirationTimestamp = event.timestamp + SUDDEN_DEATH_DURATION;
+  }
+
+  onCast(event: CastEvent) {
+    let usedForAshenJuggernaut = false;
+    let usedForSuddenDeath = false;
+    let usedForMarkedForExecution = false;
+
+    if (this.ashenJuggernautExpirationTimestamp - event.timestamp < BUFF_REFRESH_BUFFER) {
+      usedForAshenJuggernaut = true;
+    }
+    // Assume new AJ duration on cast of Execute
+    // which is true unless the execute is parried
+    this.ashenJuggernautExpirationTimestamp = event.timestamp + ASHEN_JUGGERNAUT_DURATION;
+
+    if (
+      this.suddenDeathExpirationTimestamp - event.timestamp < BUFF_REFRESH_BUFFER ||
+      this.selectedCombatant.getBuffStacks(SPELLS.SUDDEN_DEATH_FURY_TALENT_BUFF) === 2
+    ) {
+      usedForSuddenDeath = true;
+    }
+
+    if (this.markedForExecutionStacks === MAX_MARKED_FOR_EXECUTION_STACKS) {
+      usedForMarkedForExecution = true;
+    }
+
+    let rage = 0;
+    const rageResource = event.classResources?.find(
+      (resource) => resource.type === RESOURCE_TYPES.RAGE.id,
+    );
+    if (!rageResource) {
+      return;
+    } else {
+      rage = rageResource.amount / 10;
+    }
+    this.ramWasAvailable = rage > RAMPAGE_RAGE_COST;
+    this.rbWasAvailable = this.spellUsable.isAvailable(SPELLS.RAGING_BLOW.id);
+
+    if (
+      !usedForAshenJuggernaut &&
+      !usedForSuddenDeath &&
+      !usedForMarkedForExecution &&
+      (this.rbWasAvailable || this.ramWasAvailable)
+    ) {
+      this.overusedExecutes += 1;
+    }
+  }
+
+  suggestions(when: When) {
+    when(this.suggestionThresholds).addSuggestion((suggest, actual, recommended) =>
+      suggest(
+        <>
+          You cast execute prematurely {actual} times. Execute has powerful buffs and debuffs
+          associated with it, but does relatively little damage on its own. Refer to Wowhead or
+          Maxroll guides for a full description of when to best use{' '}
+          <SpellLink spell={SPELLS.EXECUTE_FURY} />
+        </>,
```

Comment:
is the maxroll guide for fury actually good? i know for a lot of specs their guides have been crap but maybe they got a good fury writer.

Reply:
Yeah, it's written (_actually_ written, not GPTd as far as I know) by Revvez, and tends to have very solid info. It sometimes takes him a bit to update after RWF, hence recommending Wowhead as well